VideoLocker/src/main/java/org/edx/mobile/module/analytics/ISegment.java

```diff
@@ -121,161 +123,167 @@ Properties trackVideoOrientation(String videoId, Double currentTime,
+        String APP = "app_name";
+        String LOADED_VIDEO = "Loaded Video";
+        String PLAYED_VIDEO = "Played Video";
+        String PAUSED_VIDEO = "Paused Video";
+        String STOPPED_VIDEO = "Stopped Video";
+        String SEEK_VIDEO = "Seeked Video";
+        String SHOW_TRANSCRIPT = "Show Transcript";
+        String HIDE_TRANSCRIPT = "Hide Transcript";
+        String VIDEO_DOWNLOADED = "Video Downloaded";
+        String BULK_DOWNLOAD_SECTION = "Bulk Download Section";
+        String BULK_DOWNLOAD_SUBSECTION = "Bulk Download Subsection";
+        String SINGLE_VIDEO_DOWNLOAD = "Single Video Download";
+        String SCREEN_TOGGLED = "Screen Toggled";
+        String USER_LOGIN = "User Login";
+        String USER_LOGOUT = "User Logout";
+        String BROWSER_LAUNCHED = "Browser Launched";
+        String LANGUAGE_CLICKED = "Language Clicked";
+        String SIGN_UP = "Sign up Clicked";
+        String FIND_COURSES = "Find Courses Clicked";
+        String CREATE_ACCOUNT_CLICKED = "Create Account Clicked";
+        String ENROLL_COURSES = "Enroll Course Clicked";
+        String EMAIL_OPT_IN = "email_opt_in";
+        String PROVIDER = "provider";
+        String BLOCK_ID = "block_id";
+        String SUPPORTED = "supported";
+        String NEW_OUTLINE_MODE = "new-mode";
+        String DEVICE_ORIENTATION = "device-orientation";
+        String NAVIGATION_MODE = "navigation-mode";
+
+        String TRACK_CELL_CONNECTION = "Cell Connection Established";
+        String CELL_CARRIER = "cell_carrier";
+        String CELL_ZERO_RATED = "cell_zero_rated";
+
+        String SPEED = "Connected Speed Report";
+        String CONNECTION_TYPE = "connection_type";
+        String CONNECTION_SPEED = "connection_speed";
+
+        String GROUP_ID = "group_id";
+        String GROUP_USER_COUNT = "group_user_count";
+        String GROUP_INVITED_USER_COUNT = "group_invited_count";
+        String SOCIAL_NETWORK = "social_network";
+        String SOCIAL_CONNECTION_STATE = "social_connection_state";
+        String SETTING_COURSES_VISIBLE_STATE = "settings_courses_visible_state";
+        String CATEGORY = "category";
+        String LABEL = "label";
+        String ACTION = "action";
+
+        String COURSE_GROUP_ACCESSED = "Course Group Accessed";
+        String GAME_GROUP_ACCESSED = "Game Group Accessed";
+        String GAME_GROUP_CREATE = "Game Group Created";
+        String GAME_GROUP_INVITE = "Game Group Invited";
+        String SOCIAL_COURSE_SHARED = "Social Course Shared";
+        String SOCIAL_CERTIFICATE_SHARED = "Social Certificate Shared";
+        String SOCIAL_CONNECTION_CHANGE = "Social Connection Change";
+        String SETTING_COURSES_VISIBLE_CHANGE = "Settings Courses Visibility Change";
+
+        String SWITCH_OUTLINE_MODE = "Switch outline mode";
+        String COMPONENT_VIEWED = "Component Viewed";
+        String OPEN_IN_BROWSER = "Browser Launched";
+        String COURSE_DASHBOARD = "Course Dashboard";
```

Comment:
Also I think we should add [`@StringDef` enum annotations](https://developer.android.com/tools/debugging/annotations.html#enum-annotations) groups for these constant variable definitions and in the parameters. That would provide the exclusivity of enums, without the memory/storage overhead. However, if that is too much work for now, than we can just merge this PR for now and defer that to later.